docs/enhancements/template.md

```diff
@@ -0,0 +1,154 @@
+---
+title: neat-enhancement-idea
+authors:
+  - "@janedoe"
+reviewers:
+  - TBD
+  - "@alicedoe"
+approvers:
+  - TBD
+  - "@oscardoe"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced|informational
+see-also:
+  - "docs/enhancements/this-other-neat-thing.md"
+replaces:
+  - "docs/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "docs/enhancements/our-past-effort.md"
+---
+
+Start by filling out this header template with metadata for this enhancement.
+
+* `reviewers`: This can be anyone that has an interest in this work.
+
+* `approvers`: All enhancements must be approved, but the appropriate people to
+  approve a given enhancement depends on its scope.  If an enhancement is
+  limited in scope to a given team or component, then a peer or lead on that
+  team or pillar is an appropriate approver.  If an enhancement captures
+  something more broad in scope, then one of the Assisted Installer architects
+  or someone they delegate would be appropriate. Use your best
+  judgement to determine the level of approval needed.  If you’re not sure,
+  just leave it blank and ask for input during review.
+
+# Neat Enhancement Idea
+
+This is the title of the enhancement. Keep it simple and descriptive. A good
+title can help communicate what the enhancement is and should be considered as
+part of any review.
+
+The YAML `title` should be lowercased and spaces/punctuation should be
+replaced with `-`.
+
+To get started with this template:
+1. **Make a copy of this template.** Copy this template.
+1. **Fill out the "overview" sections.** This includes the Summary and
+   Motivation sections. These should be easy and explain why the community
+   should desire this enhancement.
+1. **Create a PR.** Assign it to folks with expertise in that domain to help
+   sponsor the process.
+
+The `Metadata` section above is intended to support the creation of tooling
+around the enhancement process.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
```

Comment:
Isn't it a chicken-and-egg situation? That is, an enhancement proposal requires design details, a test plan, etc. which probably don't make sense to work on until/unless the proposal has been approved. A few OpenShift enhancement documents I saw looked pretty much like a detailed design. Do we want it here? I'm also afraid many people on the team won't be happy with that.

Reply:
If I understand correctly, these are not prerequisites to starting development on this feature, but things that need to be sorted out before release.

That said, this feels like it belongs in JIRA rather than our enhancement doc especially because it includes things that other teams (who may not know or care about the enhancement docs here) will need to complete. For example, test plan (QE team), operational readiness (app-sre?), documentation all feel like we would might need to create separate tickets for those teams

Reply:
I think that we said that we'll start by handling just our designs in git for now.
Maybe we should start with a lighter template that is more geared towards design documents?

Reply:
Dropping this entirely.